File: src/i18n/resolver.ts
import { en } from './en';
import { sa } from './sa';
import { hi } from './hi';
import type { PanchangTranslations } from './types';
import type { Language } from '../types/options';

const TRANSLATIONS: Record<Language, PanchangTranslations> = { en, sa, hi };

export function getTranslations(lang: Language): PanchangTranslations {
  return TRANSLATIONS[lang] ?? en;
}

export function resolveTithiName(index: number, lang: Language): string {
  const t = getTranslations(lang);
  if (index === 14) return t.misc.purnima;
  if (index === 29) return t.misc.amavasya;
  const paksha = index < 15 ? t.pakshaNames.shukla : t.pakshaNames.krishna;
  const nameIndex = index < 15 ? index : index - 15;
  return `${paksha} ${t.tithiNames[nameIndex]!}`;
}

export function resolveNakshatraName(index: number, lang: Language): string {
  return getTranslations(lang).nakshatraNames[index]!;
}

export function resolveYogaName(index: number, lang: Language): string {
  return getTranslations(lang).yogaNames[index]!;
}

export function resolveKaranaName(index: number, lang: Language): string {
  const t = getTranslations(lang);
  if (index === 0) return t.karanaNames.fixed[0]!; // Kimstughna
  if (index >= 57) return t.karanaNames.fixed[index - 56]!; // Shakuni=1, Chatushpada=2, Naga=3
  return t.karanaNames.movable[(index - 1) % 7]!;
}

export function resolveMasaName(index: number, lang: Language): string {
  return getTranslations(lang).masaNames[index]!;
}

export function resolveChandraMasaName(index: number, lang: Language, adhika = false): string {
  const t = getTranslations(lang);
  const name = t.chandraMasaNames[index]!;
  return adhika ? `${t.misc.adhika} ${name}` : name;
}

File: source/go/internal/i18n/resolver.go
package i18n

import "github.com/ishankgupta95/panchang-ts/source/go/v5/internal/types"

var translations = map[types.Language]PanchangTranslations{
	types.LanguageEn: en,
	types.LanguageHi: hi,
}

func GetTranslations(lang types.Language) PanchangTranslations {
	if t, ok := translations[lang]; ok {
		return t
	}
	return en
}

func ResolvePakshaName(index int, lang types.Language) string {
	t := GetTranslations(lang)
	if index < 15 {
		return t.PakshaNames.Shukla
	}
	return t.PakshaNames.Krishna
}

func ResolveTithiName(index int, lang types.Language) string {
	t := GetTranslations(lang)
	if index == 14 {
		return t.Misc.Purnima
	}
	if index == 29 {
		return t.Misc.Amavasya
	}
	paksha := t.PakshaNames.Krishna
	nameIndex := index - 15
	if index < 15 {
		paksha = t.PakshaNames.Shukla
		nameIndex = index
	}
	return paksha + " " + t.TithiNames[nameIndex]
}

func ResolveNakshatraName(index int, lang types.Language) string {
	return GetTranslations(lang).NakshatraNames[index]
}

func ResolveYogaName(index int, lang types.Language) string {
	return GetTranslations(lang).YogaNames[index]
}

func ResolveAnandadiYogaName(index int, lang types.Language) string {
	return GetTranslations(lang).AnandadiYogaNames[index]
}

func ResolveKaranaName(index int, lang types.Language) string {
	t := GetTranslations(lang)
	if index == 0 {
		return t.KaranaNames.Fixed[0]
	}
	if index >= 57 {
		return t.KaranaNames.Fixed[index-56]
	}
	return t.KaranaNames.Movable[(index-1)%7]
}

func ResolveMasaName(index int, lang types.Language) string {
	return GetTranslations(lang).MasaNames[index]
}

func ResolveChandraMasaName(index int, lang types.Language, adhika bool) string {
	t := GetTranslations(lang)
	name := t.ChandraMasaNames[index]
	if adhika {
		return t.Misc.Adhika + " " + name
	}
	return name
}

var AllLanguages = []types.Language{types.LanguageEn, types.LanguageHi}
